### PR TITLE
User content translation for Field Report

### PR DESCRIPTION
**Pull request by frozenhelium**
Depends on: https://github.com/IFRCGo/go-api/pull/1834

Addresses: https://github.com/IFRCGo/go-api/issues/1697

Test server: https://ifrc-go-feature-dynamic-translation.surge.sh/

### CODE REVIEW

File: src/root/components/Page/index.tsx [hunk 1]
```diff
@@ -19,6 +19,8 @@ import {
 } from '#components/NewGlobalLoading';
 
 import styles from './styles.module.scss';
+import { Language } from '#types/project';
+import TranslationWarningBanner from '#components/TranslationWarningBanner';
```
Comment (barshathakuri) [inline src/root/components/Page/index.tsx:23]: Please add imports in order

File: src/root/components/SearchSelectInput/styles.module.scss [hunk 2]
```diff
@@ -1,48 +1,51 @@
...
-
-.select {
-  :global {
-    .go__control {
-      border: 0;
-      box-shadow: unset;
-      min-height: unset;
-      background-color: unset;
+  .select {
+    margin: 0 -1*$spacing;
```
Comment (barshathakuri) [inline src/root/components/SearchSelectInput/styles.module.scss:5]: Please try not to use negative margins

File: src/root/components/ThreeWForm/index.tsx [hunk 3]
```diff
@@ -55,6 +58,7 @@ import {
 import AnnualSplitInput from './AnnualSplitInput';
 
 import styles from './styles.module.scss';
+import Container from '#components/Container';
```
Comment (barshathakuri) [inline src/root/components/ThreeWForm/index.tsx:61]: Same as above.

File: src/root/components/ThreeWForm/index.tsx [hunk 4]
```diff
@@ -148,6 +158,9 @@ function ThreeWForm(props: Props) {
     submitRequest(finalValues);
   }, [onValueSet, submitRequest]);
 
+  // const languageMismatch = (isDefined(projectId) && fieldReportResponse?.translation_module_original_language !== currentLanguage) ?? false;
```
Comment (barshathakuri) [inline src/root/components/ThreeWForm/index.tsx:161]: Please remove this

File: src/root/components/ThreeWForm/index.tsx [hunk 5]
```diff
@@ -348,447 +361,459 @@ function ThreeWForm(props: Props) {
...
-                <button
-                  title={strings.projectFormSelectAllDistricts}
-                  type="button"
-                  className={_cs(
-                    styles.selectAllDistrictsButton,
-                    'button button--secondary',
-                    shouldDisableDistrictInput && 'disabled',
-                  )}
+          {languageMismatch && projectDetailsResponse && (
+            <Container contentClassName={styles.languageMismatch}>
```
Comment (barshathakuri) [inline src/root/components/ThreeWForm/index.tsx:365]: Please break this container and styling in different line :
<Container 
contentClassName={styles.languageMismatch}
>

File: src/root/components/ThreeWForm/styles.module.scss [hunk 6]
```diff
@@ -1,6 +1,10 @@
 @import '~styles/settings/variables';
 
 .three-w-form {
+  .languageMismatch {
```
Comment (barshathakuri) [inline src/root/components/ThreeWForm/styles.module.scss:4]: Please write languageMismatch in scss format (language-mismatch)

File: src/root/components/ThreeWForm/styles.module.scss [hunk 7]
```diff
@@ -25,8 +29,4 @@
   .disable {
     color: grey
```
Comment (barshathakuri) [inline src/root/components/ThreeWForm/styles.module.scss:30]: Please use styling variables

File: src/root/components/TranslationWarningBanner/index.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { _cs } from '@togglecorp/fujs';
+
+import { languageOptions } from '#utils/lang';
+import Translate from '#components/Translate';
+import useReduxState from '#hooks/useReduxState';
+import styles from './styles.module.scss';
```
Comment (barshathakuri) [inline src/root/components/TranslationWarningBanner/index.tsx:7]: Same as above

File: src/root/views/DrefApplicationForm/index.tsx [hunk 9]
```diff
@@ -61,6 +61,9 @@ import {
 } from './import';
 
 import styles from './styles.module.scss';
+import useReduxState from '#hooks/useReduxState';
+import { languageOptions } from '#utils/lang';
+import Translate from '#components/Translate';
```
Comment (barshathakuri) [inline src/root/views/DrefApplicationForm/index.tsx:66]: Same as above

File: src/root/views/DrefApplicationForm/styles.module.scss [hunk 10]
```diff
@@ -32,3 +32,7 @@
       margin: 0;
     }
 }
+
+.languageMismatch {
```
Comment (barshathakuri) [inline src/root/views/DrefApplicationForm/styles.module.scss:36]: Use scss format

File: src/root/views/DrefOperationalUpdateForm/index.tsx [hunk 11]
```diff
@@ -62,6 +62,9 @@ import Submission from './Submission';
 import ObsoletePayloadResolutionModal from './ObsoletePayloadResolutionModal';
 
 import styles from './styles.module.scss';
+import Translate from '#components/Translate';
+import useReduxState from '#hooks/useReduxState';
+import { languageOptions } from '#utils/lang';
```
Comment (barshathakuri) [inline src/root/views/DrefOperationalUpdateForm/index.tsx:67]: Same as above

File: src/root/views/DrefOperationalUpdateForm/index.tsx [hunk 12]
```diff
@@ -714,123 +721,135 @@ function DrefOperationalUpdate(props: Props) {
...
-                    onValueChange={onValueChange}
-                    value={value}
-                    yesNoOptions={yesNoOptions}
-                    needOptions={needOptions}
-                    nsActionOptions={nsActionOptions}
-                    fileIdToUrlMap={fileIdToUrlMap}
-                    setFileIdToUrlMap={setFileIdToUrlMap}
-                    drefType={drefType}
+              {languageMismatch && drefOperationalResponse && (
+                <Container contentClassName={styles.languageMismatch}>
```
Comment (barshathakuri) [inline src/root/views/DrefOperationalUpdateForm/index.tsx:725]: Same as above

File: src/root/views/DrefOperationalUpdateForm/styles.module.scss [hunk 13]
```diff
@@ -19,3 +19,7 @@
   animation: flash $duration-animation-fast 5 ease-in-out;
   animation-delay: $duration-animation-medium;
 }
+
+.languageMismatch {
```
Comment (barshathakuri) [inline src/root/views/DrefOperationalUpdateForm/styles.module.scss:23]: Same as above

File: src/root/views/FinalReportForm/styles.module.scss [hunk 14]
```diff
@@ -9,4 +9,8 @@
   >* {
     margin: 0 $spacing*0.5;
   }
-}
\ No newline at end of file
+}
+
+.languageMismatch {
+  text-align: center;
```
Comment (barshathakuri) [inline src/root/views/FinalReportForm/styles.module.scss:15]: Same

File: src/root/views/FlashUpdateApplicationForm/styles.module.scss [hunk 15]
```diff
@@ -12,14 +12,18 @@
...
+
+  p {
+    margin: 0;
+  }
+}
 
-    p {
-      margin: 0;
-    }
+.languageMismatch {
```
Comment (barshathakuri) [inline src/root/views/FlashUpdateApplicationForm/styles.module.scss:27]: Same

**Review by barshathakuri — CHANGES_REQUESTED** [review]

**Review by barshathakuri — APPROVED** [review]

File: src/root/components/ThreeWDetails/index.tsx [hunk 16]
```diff
@@ -144,7 +144,6 @@ function ProjectDetail(props: Props) {
               value={get(projectResponse, 'description', false)}
               style={{backgroundColor: '#faf9f9', padding: '1.5rem'}}
```
Comment (puranban) [inline src/root/components/ThreeWDetails/index.tsx:145]: avoid inline css

File: src/root/components/ThreeWForm/index.tsx [hunk 17]
```diff
@@ -348,447 +358,461 @@ function ThreeWForm(props: Props) {
...
-                value={value.event}
-                onChange={onValueChange}
-              />
-            </InputSection>
-          )}
-          { shouldShowCurrentEmergencyOperation && (
-            <InputSection
-              title={strings.projectFormCurrentEmergency}
-              description={strings.projectFormCurrentEmergencyHelpText}
+          {languageMismatch && projectDetailsResponse && (
```
Comment (puranban) [inline src/root/components/ThreeWForm/index.tsx:361]: no need to recheck `projectDetailsResponse` value

File: src/root/components/ThreeWForm/index.tsx [hunk 18]
```diff
@@ -111,6 +117,8 @@ function ThreeWForm(props: Props) {
     },
   });
 
+  const languageMismatch = (isDefined(projectId) && projectDetailsResponse?.translation_module_original_language !== currentLanguage) ?? false;
+
```
Comment (puranban) [inline src/root/components/ThreeWForm/index.tsx:121]: const languageMismatch = isTruthy(projectDetailsResponse?.translation_module_original_language !== currentLanguage);

NOTE: use `isTruthy` from fujs.

File: src/root/components/TranslationWarningBanner/index.tsx [hunk 19]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { _cs } from '@togglecorp/fujs';
+
+import { languageOptions } from '#utils/lang';
+import useReduxState from '#hooks/useReduxState';
+
+import Translate from '#components/Translate';
```
Comment (puranban) [inline src/root/components/TranslationWarningBanner/index.tsx:7]: Apply the same import rule as requested in the above comment.

File: src/root/views/DrefApplicationForm/index.tsx [hunk 20]
```diff
@@ -337,6 +343,8 @@ function DrefApplication(props: Props) {
     }
   });
 
+  const languageMismatch = (isDefined(drefId) && drefResponse?.translation_module_original_language !== currentLanguage) ?? false;
+
```
Comment (puranban) [inline src/root/views/DrefApplicationForm/index.tsx:347]: use `isTruthy` here too.

File: src/root/views/FieldReportForm/index.tsx [hunk 21]
```diff
@@ -312,7 +307,7 @@ function FieldReportForm(props: Props) {
 
       setCurrentStep(nextStepMap[currentStep]);
     }
-  }, [submitRequest, userDetails, currentStep, setCurrentStep, validate, onErrorSet,countryIsoOptions, disasterTypeOptions, eventOptions, strings.fieldReportCOVID19, value.country, value.event, reportId]);
+  }, [submitRequest, userDetails, currentStep, setCurrentStep, validate, onErrorSet, countryIsoOptions, disasterTypeOptions, eventOptions, strings.fieldReportCOVID19, value.country, value.event, reportId]);
```
Comment (puranban) [inline src/root/views/FieldReportForm/index.tsx:310]: use linebreak when there are more than three dependencies.

**Review by puranban — CHANGES_REQUESTED** [review]
Please do not use auto-formatting.

**Comment by frozenhelium** [comment]
@szabozoltan69 lets merge and deploy this after https://github.com/IFRCGo/go-api/pull/1834

**Comment by szabozoltan69** [comment]
@frozenhelium I'll do the merges  (both api and frontend), despite of the fact that locally I could not use these new translation branches. The frontend gave some kind of CORS error.

**Comment by szabozoltan69** [comment]
@frozenhelium @thenav56 Do we have to set some env variable for the frontend change of translations?

**Comment by szabozoltan69** [comment]
@frozenhelium @thenav56 Both backend and frontend are deployed to Staging.

**Pull request merged**